### PR TITLE
Add support for filtering parents by has_many associations

### PR DESCRIPTION
**Pull request by carllerche**
## Summary
This PR adds support for filtering parent models by conditions on their `has_many` associations using a new `.any()` method. This enables queries like "find users who have at least one incomplete todo".

## Key Changes

- **New `.any()` method on collection paths**: Added `Path<[T]>::any()` in `stmt/path.rs` that builds an `IN subquery` expression to test whether any associated record satisfies a filter condition.

- **Code generation for `.any()` method**: Updated `expand/relation.rs` to generate the `.any()` method on `HasMany` field accessors, allowing ergonomic filtering syntax like `User::fields().todos().any(Todo::fields().complete().eq(false))`.

- **Extended subquery lifting logic**: Modified `lift_in_subquery.rs` to handle `HasMany` associations (in addition to existing `BelongsTo` and `HasOne` support) by:
  - Refactoring `lift_has_one_in_subquery()` into a more general `lift_has_n_in_subquery()` method
  - Properly transforming has_many/has_one paths into foreign-key-based `IN` subqueries
  - Extracting the correct foreign key field references for the subquery

- **Comprehensive integration tests**: Added `has_many_filter_on_association.rs` with two test cases:
  - `filter_parent_by_child_field`: Tests filtering users by incomplete/complete todos
  - `filter_parent_no_matching_children`: Tests edge case where no children match the filter

## Implementation Details

The implementation transforms conceptual queries like:
```
User.todos IN (SELECT * FROM Todo WHERE complete = false)
```

Into efficient SQL:
```sql
User.id IN (SELECT Todo.user_id FROM Todo WHERE complete = false)
```

This approach reuses the existing subquery lifting infrastructure and maintains consistency with how `BelongsTo` and `HasOne` associations are handled.

https://claude.ai/code/session_01Hit4N8y4PDmsvjNnH53WVi

### CODE REVIEW

File: crates/toasty/src/engine/simplify/lift_in_subquery.rs [hunk 1]
```diff
@@ -130,38 +137,43 @@ impl Simplify<'_> {
...
         match &mut subquery.body {
-            stmt::ExprSet::Select(subquery) => {
-                subquery.returning = stmt::Returning::Expr(match &pair.foreign_key.fields[..] {
-                    [fk_field] => stmt::Expr::ref_self_field(fk_field.source),
-                    _ => todo!("composite key"),
-                });
+            stmt::ExprSet::Select(select) => {
+                select.returning = stmt::Returning::Expr(stmt::Expr::ref_self_field(child_field));
             }
             _ => todo!(),
```
Comment (martin-g) [inline crates/toasty/src/engine/simplify/lift_in_subquery.rs]: nit:
```suggestion
            unsupported => todo!("Unsupported expression: {unsupported:?}"),
```
for easier debugging if it ever fails

File: crates/toasty-driver-integration-suite/src/tests/has_many_filter_on_association.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+        complete: bool,
+    }
+
+    let mut db = test.setup_db(models!(User, Todo)).await;
+
+    // Create users
+    let alice = User::create().name("Alice").exec(&mut db).await?;
+    let bob = User::create().name("Bob").exec(&mut db).await?;
+    let carol = User::create().name("Carol").exec(&mut db).await?;
```
Comment (martin-g) [inline crates/toasty-driver-integration-suite/src/tests/has_many_filter_on_association.rs]: Maybe also add a user without any todos

File: crates/toasty/src/stmt/path.rs [hunk 3]
```diff
@@ -132,6 +132,27 @@ impl<T: ?Sized> Path<T> {
...
+    /// parent query.
+    pub fn any(self, filter: Expr<bool>) -> Expr<bool>
+    where
+        T: super::Model,
+    {
+        // Build a query on the child model filtered by `filter`
+        let child_query = super::Select::<T>::filter(filter);
+
+        Expr {
+            untyped: stmt::Expr::in_subquery(self.untyped.into_stmt(), child_query.untyped),
```
Comment (martin-g) [inline crates/toasty/src/stmt/path.rs]: My AI friend says that if `.any()` is called on a chained association path then `self.untyped.into_stmt()` is `Expr::Project` and it would fail at https://github.com/tokio-rs/toasty/pull/447/changes#diff-12ab520f0399b2c84c4bcc4e46c3003b9e9d0cd9cd26a8c7bbb92c8bb14e9f17R25
